enso/experiment/__init__.py

```diff
@@ -72,14 +71,30 @@ def run_experiments(self):
             except Exception:
                 logging.exception("Exception occurred for {}".format(current_setting))
 
+    def experiment_has_been_run(self, current_settings):
```

Comment:
Just a note that this could be kind of an expensive check once these files get a bit longer.  I think it's necessary functionality though.